src/main/java/seedu/address/model/profile/ProfileItem.java

```diff
@@ -0,0 +1,102 @@
+package seedu.address.model.profile;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a Profile Item in the address book.
+ * Guarantees: details are present and not null, field values are validated, immutable.
+ */
+public class ProfileItem {
+
+    // Identity fields
+    private final String title;
+    private final String type;
+
+    // Data fields
+    private final Set<Tag> tags = new HashSet<>();
+
+    /**
+     * Every field must be present and not null.
+     */
+    public ProfileItem(String title, String type, Set<Tag> tags) {
+        requireAllNonNull(type, title, tags);
+        this.type = type;
+        this.title = title;
+        this.tags.addAll(tags);
+    }
+
+
+    public String getTitle() {
+        return title;
+    }
+
+    public String getType() {
+        return type;
+    }
+    /**
+     * Returns an immutable tag set, which throws {@code UnsupportedOperationException}
+     * if modification is attempted.
+     */
+    public Set<Tag> getTags() {
+        return Collections.unmodifiableSet(tags);
+    }
+
+    /**
+     * Returns true if both profiles of the same name have at least one other identity field that is the same.
+     * This defines a weaker notion of equality between two persons.
```

Comment:
should be two profile items? and several comments below

Reply:
I stopped changing this because I was unsure if we need 2 levels of inequality for profile items.

Reply:
@seanjyjy @orzymandias @ZoroarkDarkrai @shawn-nyk 
I think this is a good topic for discussion. Currently, AB3 only uses the weaker notion of equality, `isSameItem` for the edit command. All the other commands check based on the `equals` method. Do we want to change it such that the weaker notion of equality also applies to all the other commands as well?